lib/test_data/nsm_builder.rb

```diff
@@ -0,0 +1,86 @@
+module TestData
+  class NsmBuilder
+    def build_many(bulk: 100, large: 4, year: 2023)
+      raise 'Do not run on production' if HostEnv.production?
+
+      bulk.times do
+        build(year:)
+
+        # avoid issues with large number of applications with the same last_updated_at time
+        sleep 0.1 unless HostEnv.local?
```

Comment:
Why is this not an issue locally?

Reply:
so this is mostly to avoid the delay in the tests. I will change it to `Rails.env.test?` so this is more clear